templates/agent/worker/simple/getTldrawAgentChangesFromSimpleEvents.ts

```diff
@@ -0,0 +1,786 @@
+import { TLAiSerializedPrompt } from '@tldraw/ai'
+import {
+	IndexKey,
+	TLArrowShape,
+	TLDefaultFillStyle,
+	TLGeoShape,
+	TLLineShape,
+	TLNoteShape,
+	TLRichText,
+	TLShape,
+	TLShapeId,
+	TLShapePartial,
+	TLTextShape,
+	toRichText,
+} from 'tldraw'
+import { Streaming, TLAgentChange } from '../../client/AgentChange'
+import {
+	ISimpleColor,
+	ISimpleCreateEvent,
+	ISimpleDeleteEvent,
+	ISimpleEvent,
+	ISimpleFill,
+	ISimpleLabelEvent,
+	ISimpleMoveEvent,
+	ISimpleUpdateEvent,
+	SimpleColor,
+} from './schema'
+
+function toRichTextIfNeeded(text: string | { type: string; content: any[] }): TLRichText {
+	if (typeof text === 'string') {
+		return toRichText(text)
+	}
+	return text
+}
+
+export function getTldrawAgentChangesFromSimpleEvents(
+	prompt: TLAiSerializedPrompt,
+	event: Streaming<ISimpleEvent>
+): Streaming<TLAgentChange>[] {
+	if (event.complete) {
+		console.log('getTldrawAiChangesFromSimpleEvents [EVENT FROM MODEL]', event)
+	}
+
+	switch (event._type) {
+		case 'update': {
+			return getTldrawAiChangesFromSimpleUpdateEvent(prompt, event)
+		}
+		case 'create': {
+			return getTldrawAiChangesFromSimpleCreateEvent(prompt, event)
+		}
+		case 'label': {
+			return getTldrawAiChangesFromSimpleLabelEvent(prompt, event)
+		}
+		case 'delete': {
+			return getTldrawAiChangesFromSimpleDeleteEvent(prompt, event)
+		}
+		case 'move': {
+			return getTldrawAiChangesFromSimpleMoveEvent(prompt, event)
+		}
+		case 'message': {
+			const { _type, ...change } = event
+			return [{ ...change, type: _type }]
+		}
+		case 'think': {
+			const { _type, ...change } = event
+			return [{ ...change, type: _type }]
+		}
+		case 'schedule': {
+			const { _type, ...change } = event
+			return [{ ...change, type: _type }]
+		}
+		case 'setMyView': {
+			const { _type, ...change } = event
+			return [{ ...change, type: _type }]
+		}
+		default: {
+			return [{ type: 'raw', event, complete: false }]
+		}
+	}
+}
+
+const FILL_MAP: Record<ISimpleFill, TLDefaultFillStyle> = {
+	none: 'none',
+	solid: 'fill',
+	semi: 'semi',
+	tint: 'solid',
+	pattern: 'pattern',
+}
```

Comment:
We should move and/or merge this and the `getTldrawAiChangesFromSimpleUpdateEvent` functions into conversions.ts